src/meta/src/hummock/error.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::error::{ErrorCode, RwError, ToErrorStr};
+use risingwave_hummock_sdk::HummockContextId;
+use thiserror::Error;
+
+use crate::storage::meta_store;
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("invalid hummock context {0}")]
+    InvalidContext(HummockContextId),
+    #[error(transparent)]
+    MetaStoreError(anyhow::Error),
+    #[error("internal error: {0}")]
+    InternalError(String),
+}
+
+impl Error {
+    pub fn retryable(&self) -> bool {
+        match self {
+            Error::InvalidContext(_) => false,
+            Error::MetaStoreError(_) => true,
+            Error::InternalError(_) => false,
+        }
+    }
+}
+
+impl From<meta_store::Error> for Error {
+    fn from(error: meta_store::Error) -> Self {
+        match error {
+            meta_store::Error::ItemNotFound(err) => Error::InternalError(err),
+            meta_store::Error::TransactionAbort() => {
+                // TODO: need more concrete error from meta store.
+                Error::InvalidContext(0)
+            }
+            meta_store::Error::Internal(err) => Error::MetaStoreError(err),
```

Comment:
Why do we convert Internal error to `MetaStoreError` (retryable)? Converntionally, internal error is normally not retryable.

Reply:
Currently `meta_store::Error::Internal` is equivalent to `EtcdError`. Need to expand `meta_store::Error::Internal` to more detail meta_store errors.